velox/connectors/hive/storage_adapters/abfs/AzureClientProviders.cpp

```diff
@@ -0,0 +1,238 @@
+    const std::shared_ptr<AbfsPath>& abfsPath,
+    const config::ConfigBase& config) {
+  auto credKey =
+      fmt::format("{}.{}", kAzureAccountKey, abfsPath->accountNameWithSuffix());
+  VELOX_USER_CHECK(config.valueExists(credKey), "Config {} not found", credKey);
+  auto firstDot = abfsPath->accountNameWithSuffix().find_first_of(".");
+  auto endpointSuffix =
+      abfsPath->accountNameWithSuffix().substr(firstDot + 5 /* .dfs. */);
+  std::stringstream ss;
+  ss << "DefaultEndpointsProtocol=" << (abfsPath->isHttps() ? "https" : "http");
+  ss << ";AccountName=" << abfsPath->accountName();
+  ss << ";AccountKey=" << config.get<std::string>(credKey).value();
+  ss << ";EndpointSuffix=" << endpointSuffix;
+
+  if (config.valueExists(kAzureBlobEndpoint)) {
+    ss << ";BlobEndpoint="
+       << config.get<std::string>(kAzureBlobEndpoint).value();
+  }
+  ss << ";";
+  connectionString_ = ss.str();
+}
+
+std::unique_ptr<AzureBlobClient> OAuthAzureClientProvider::getReadFileClient(
+    const std::shared_ptr<AbfsPath>& abfsPath,
+    const config::ConfigBase& config) {
+  init(abfsPath, config);
+  const auto url = abfsPath->getUrl(true);
+  auto client = std::make_unique<BlobClient>(url, tokenCredential_);
+  return std::make_unique<BlobClientWrapper>(std::move(client));
+}
+
+std::unique_ptr<AzureDataLakeFileClient>
+OAuthAzureClientProvider::getWriteFileClient(
+    const std::shared_ptr<AbfsPath>& abfsPath,
+    const config::ConfigBase& config) {
+  init(abfsPath, config);
+  const auto url = abfsPath->getUrl(false);
+  auto client = std::make_unique<DataLakeFileClient>(url, tokenCredential_);
+  return std::make_unique<DataLakeFileClientWrapper>(std::move(client));
+}
+
+std::pair<std::string, std::string>
+OAuthAzureClientProvider::tenantIdAndAuthorityHost(
+    const std::shared_ptr<AbfsPath>& abfsPath,
+    const config::ConfigBase& config) {
+  init(abfsPath, config);
+  return {tenentId_, authorityHost_};
+}
+
+void OAuthAzureClientProvider::init(
+    const std::shared_ptr<AbfsPath>& abfsPath,
+    const config::ConfigBase& config) {
+  auto clientIdKey = fmt::format(
+      "{}.{}", kAzureAccountOAuth2ClientId, abfsPath->accountNameWithSuffix());
+  auto clientSecretKey = fmt::format(
+      "{}.{}",
+      kAzureAccountOAuth2ClientSecret,
+      abfsPath->accountNameWithSuffix());
+  auto clientEndpointKey = fmt::format(
+      "{}.{}",
+      kAzureAccountOAuth2ClientEndpoint,
+      abfsPath->accountNameWithSuffix());
+  VELOX_USER_CHECK(
+      config.valueExists(clientIdKey), "Config {} not found", clientIdKey);
+  VELOX_USER_CHECK(
+      config.valueExists(clientSecretKey),
+      "Config {} not found",
+      clientSecretKey);
+  VELOX_USER_CHECK(
+      config.valueExists(clientEndpointKey),
+      "Config {} not found",
+      clientEndpointKey);
+  auto clientEndpoint = config.get<std::string>(clientEndpointKey).value();
+  auto firstSep = clientEndpoint.find_first_of("/", /* https:// */ 8);
```

Comment:
nit: use the /\*argName=\*/value format (note the lack of spaces).

Reply:
This inline comment is not an arg name.